File: C++ solutions/225. Implement Stack using Queues.cpp
class MyStack {
public:
    queue<int>q;
    MyStack() {
       
    }
    
    void push(int x) {
        queue<int>q2;
        q2.push(x);
        while(!q.empty()){
            q2.push(q.front());
            q.pop();
        }
        swap(q,q2);
    }
    
    int pop() {
        int pop=q.front();
        q.pop();
        return pop;
    }
    
    int top() {
        return q.front();
    }
    
    bool empty() {
        return q.empty();
    }
};

/**
 * Your MyStack object will be instantiated and called as such:
 * MyStack* obj = new MyStack();
 * obj->push(x);
 * int param_2 = obj->pop();
 * int param_3 = obj->top();
 * bool param_4 = obj->empty();
 */

File: JavaScript solutions/225. Implement Stack using Queues.js
var MyStack = function () {
  this.q = [];
};

/**
 * @param {number} x
 * @return {void}
 */
MyStack.prototype.push = function (x) {
  var q2 = [x];
  while (this.q.length > 0) {
    q2.push(this.q.shift());
  }
  this.q = q2;
};

/**
 * @return {number}
 */
MyStack.prototype.pop = function () {
  return this.q.shift();
};

/**
 * @return {number}
 */
MyStack.prototype.top = function () {
  return this.q[0];
};

/**
 * @return {boolean}
 */
MyStack.prototype.empty = function () {
  return this.q.length === 0;
};

/**
 * Your MyStack object will be instantiated and called as such:
 * var obj = new MyStack()
 * obj.push(x)
 * var param_2 = obj.pop()
 * var param_3 = obj.top()
 * var param_4 = obj.empty()
 */
